submissions/BlueLamantine/friends app/js/main.js

```diff
@@ -0,0 +1,329 @@
+        }
+      });
+    });
+
+    const sort = {
+      'sort-gender': () => this.sortByGender(selected.id),
+      'sort-name': () => {
+        this.unckeckedRadioButton('sort-age');
+        this.sortByContent('.first-name', selected.id, this.sortByNameTypes);
+      },
+      'sort-age': () => {
+        this.unckeckedRadioButton('sort-name');
+        this.sortByContent('.age', selected.id, this.sortByAgeTypes);
+      },
+      default: () => false,
+    };
+    return (sort[selected.name] || sort['default'])();
+  }
+
+  startFilter() {
+    this.filtersPanel.addEventListener('click', ({ target }) => {
+      if (target.type === 'radio' || target.type === 'text') {
+        loadMoreData.loadDataByScroll(true);
+        this.selectSort(target);
+      }
+    });
+  }
+
+  sortByGender(key) {
+    const cards = document.querySelectorAll('.user-card');
+    if (key === 'allGender') {
+      Array.from(cards).map(node => node.classList.remove('hidden'));
+    } else {
+      Array.from(cards).find(node => {
+        const currentGender = node.querySelector('.gender');
+        if (currentGender.textContent !== key) {
+          currentGender.closest('.user-card').classList.add('hidden');
+        } else {
+          currentGender.closest('.user-card').classList.remove('hidden');
+        }
+      });
+    }
+  }
+
+  insertSortedProfiles(sortedCards, param) {
+    sortedCards.forEach(card =>
+      this.profilesContainer.insertBefore(card, this.profilesContainer[param])
+    );
+  }
+
+  sortBySelector(selectorName) {
+    const cards = document.querySelectorAll('.user-card');
+    if (selectorName === 'all') {
+      return Array.from(cards).sort((a, b) => a.id.localeCompare(b.id));
+    }
+    return Array.from(cards).sort((a, b) =>
+      a
+        .querySelector(selectorName)
+        .textContent.localeCompare(b.querySelector(selectorName).textContent)
+    );
+  }
+
+  sortByContent(sortSelector, sortType, sortTypes) {
+    const sortedCards = this.sortBySelector(sortSelector);
+    const typeOfSort = {
+      [sortTypes.type1]: () =>
+        this.insertSortedProfiles(sortedCards, 'lastChild'),
+      [sortTypes.type2]: () =>
+        this.insertSortedProfiles(sortedCards, 'firstChild'),
+      [sortTypes.type3]: () =>
+        this.insertSortedProfiles(this.sortBySelector('all'), 'lastChild'),
+      default: () => false,
+    };
+    return (typeOfSort[sortType] || typeOfSort['default'])();
+  }
+}
+
+class Api {
+  constructor() {
+    this.apiURL = `https://randomuser.me/api/`;
+    this.errorMessage = `Uh oh, something has gone wrong. Please tweet us @randomapi about the issue. Thank you.`;
+  }
+
+  async getData(pageNum = 1) {
+    try {
+      const response = await fetch(
+        this.apiURL + `?page=${pageNum}&results=10&seed=abc`
+      );
+      if (!response.ok) throw new Error(errorMessage);
+      const data = await response.json();
+      new Render(data.results, data.info).getProfiles();
```

Comment:
Ok, you are recreating this class only to call its methods.

And class itself is stateless.

So you can replace it with singleton and pass instance as an argument to the constructor using dependency injection.

Reply:
I feel like this ```new Users(new Render(), new Filter())``` might be the one. (dependency injection)

Reply:
But then it will be not singleton.

How many filters you have in your app?

Reply:
Ok, I got it. My goal is a have single instance of class and use it every time I need. 
So I realized singleton of ```Filter``` and ```Render``` classes with this construction :
```class ClassName {
constructor() {

    if (ClassName.instance == null) {
    ...
     ClassName.instance = this;
    }
    return ClassName.instance;
}